plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2212,14 +2212,13 @@ public TableStatisticsMetadata getStatisticsCollectionMetadataForWrite(Connector
         return getStatisticsCollectionMetadata(
                 tableMetadata.getColumns(),
                 analyzeColumnNames.orElse(allColumnNames),
-                // File modified time does not need to be collected as a statistics because it gets derived directly from files being written
-                false);
+                true);
     }
 
     private TableStatisticsMetadata getStatisticsCollectionMetadata(
             List<ColumnMetadata> tableColumns,
             Set<String> analyzeColumnNames,
-            boolean includeMaxFileModifiedTime)
+            boolean isCollectionOnWrite)
```

Comment:
Do we have to know in the context of this method about "write" ?
I'd argue that the previous form was more suited for the purpose of this method.

Reply:
This was introduced when we started updating statistics on write. In that context `maxFileModifiedTime` statistics needed to be skipped as files were not available and collecting statistics was failing. We did not needed them anyway as we could derive them directly from `dataFileInfos`.  
Generally this flag set to true indicated that we create statistics metadata during write (so we don't want to collect `maxFileModifiedTime`). 
With this PR we also need information if metadata is created during write or not (as we cannot collect `$path` statistics in case of write as file does not exist yet).
I think that we don't need two flags here as both statistics `maxFileModifiedTime` and `$path` cannot be collected during write.